File: src/main/java/novemberizing/ds/on/Pair.java
package novemberizing.ds.on;

import novemberizing.ds.On;

/**
 *
 * @author novemberizing, me@novemberizing.net
 * @since 2017. 1. 17.
 */
public interface Pair<A, B> extends On { void on(A first, B second); }

File: src/main/c++/novemberizing.old.3/ds/tuple/pair.cc
#include "pair.hh"


#ifdef    __NOVEMBERIZING_DS_TUPLE__PAIR__EXAMPLE__

#include <iostream>

using namespace novemberizing;

int main(int argc, char ** argv) {

    ds::tuple::Pair<int, int> pair(1, 2);

    std::cout << pair.first << ":" << pair.second << std::endl;

    return 0;
}

#endif // __NOVEMBERIZING_DS_TUPLE__PAIR__EXAMPLE__
